frontend/__tests__/components/buttons.test.tsx

```diff
@@ -0,0 +1,99 @@
+import { fireEvent, render } from '@testing-library/react';
+
+import { createRemixStub } from '@remix-run/testing';
+
+import { describe, expect, it, vi } from 'vitest';
+
+import { Button, ButtonLink } from '~/components/buttons';
+
+describe('Button Component', () => {
+  it('renders button with default props', () => {
+    const { getByText } = render(<Button>Click me</Button>);
+    const button = getByText('Click me');
+    expect(button).toBeInTheDocument();
+    expect(button.tagName).toBe('BUTTON');
```

Comment:
You should use `.toEqual()` in every case instead of `.toBe()`

Reply:
https://vitest.dev/api/expect#tobe
> `toBe` can be used to assert if primitives are equal or that objects share the same reference. It is equivalent of calling `expect(Object.is(3, 3)).toBe(true)`. If the objects are not the same, but you want to check if their structures are identical, you can use [toEqual](https://vitest.dev/api/expect#toequal).

Reply:
Just always use `.toEqual()` since it works with both primitives *and* objects.

When you selectively use `.toBe()`, future developers can potentially copy/paste it into their tests without understanding the subtle difference. Using `.toEqual()` is always the safer choice, for current and future generations.

Reply:
`.toBe()` should be used in the majority of cases... `.toEqual()` should be use if you want to compare object. Devs should know the difference and if they not then they need to read the documentation.

Reply:
Sorry, but I'm being stubborn about this one..

This is in reference to Jest, but still applies to Vitest.

<https://dev.to/thejaredwilcurt/why-you-should-never-use-tobe-in-jest-48ca>
![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/103e8e1c-f6da-4cc0-a57d-93e08b5df12a)

Reply:
Changed to `.toEqual()`